pkg/storage/stores/shipper/compactor/compactor.go

```diff
@@ -67,6 +67,7 @@ type Config struct {
 	RetentionEnabled          bool            `yaml:"retention_enabled"`
 	RetentionDeleteDelay      time.Duration   `yaml:"retention_delete_delay"`
 	RetentionDeleteWorkCount  int             `yaml:"retention_delete_worker_count"`
+	DeletionEnabled           bool            `yaml:"deletion_enabled"`
```

Comment:
I think this should be `DeletionMode`. As I could foresee quite a few combinations in the future, as we are progression deletion. For example: `disabled|filter-only|filter-and-delete`

Reply:
Changed to DeletionMode. A v1 mode was added as well for the current functionality. That can be removed once the new delete functionality is available.